packages/SymfonyEventDispatcher/src/Adapter/Nette/DI/SymfonyEventDispatcherExtension.php

```diff
@@ -47,12 +46,38 @@ private function isKdybyEventsRegistered(): bool
 
     private function addSubscribersToEventDispatcher(): void
     {
-        DefinitionCollector::loadCollectorWithType(
-            $this->getContainerBuilder(),
-            EventDispatcherInterface::class,
-            EventSubscriberInterface::class,
-            'addSubscriber'
-        );
+        $containerBuilder = $this->getContainerBuilder();
+        $dispatcherDefinition = $containerBuilder->getDefinitionByType(EventDispatcherInterface::class);
+        $subscriberDefinitions = $containerBuilder->findByType(EventSubscriberInterface::class);
+
+        foreach ($subscriberDefinitions as $name => $definition) {
+            $this->registerSubscriber($dispatcherDefinition, $name, $definition->getClass());
+        }
+    }
+
+    private function registerSubscriber(ServiceDefinition $dispatcher, string $service, string $class): void
+    {
+        foreach ($class::getSubscribedEvents() as $event => $listeners) {
+            if (is_string($listeners)) {
+                $listeners = [[$listeners]];
+            } elseif (is_string($listeners[0])) {
+                $listeners = [$listeners];
+            }
+            foreach ($listeners as $listener) {
+                $priority = $listener[1] ?? 0;
+                $method = $listener[0];
+                $dispatcher->addSetup(
```

Comment:
Could you please move this to standalone method with descriptive naming?
So we know, what is going on here. I get lost easily in neon magic.